### DISCUSSION TITLE
Slider: how is the value displayed without an input, and other questions

### DISCUSSION THREAD

**Discussion by mbgower** [carbon-design-system/carbon#11581] [Help]
During a review of the Slider component, a number of considerations for the slider were raised. They are posted as comments in [slider code](https://ibm.box.com/s/f0fiio2et70i375cyslrp3uga09r928b) and [slider usage](https://ibm.box.com/s/yu04q63kdt8asowzkokkxcp3or5xxigw), but I will point out a question here:

There are two main variants of a slider listed -- one with a text input and one without one. Why aren't they listed as variants. Their behaviour is different. For instance, when there is no numeric input, the value of the slider is impossible to determine.

<img width="364" alt="image" src="https://user-images.githubusercontent.com/14298245/172704414-e1b740b4-a974-4d00-9f5f-1cc8c69f43eb.png">

 On the [code page](https://carbondesignsystem.com/components/slider/code), I activated the hideTextInput modifier so that the slider displays without a text input (which is allowed). What is the slider value? No way of knowing, since the input was the only label for it. And yet, we have labels at the termini. This seems problematic. 
On the Usage page it states:
> The basic slider does not include discrete values, as the slider represents a
percentage of 0-100. In this case it is not necessary for a user to choose a speci=c
value, but instead generally increase or decrease an input. For example, the user
increases the slider amount and the volume of the music gets louder.

I kinda get the idea of a range with no labels at all, but once there are labels on the ends, shouldn't the value should be shown, ideally with a persistent label, but as a fallback, on both hover and focus?

**Comment by laurenmrice** [comment]
@mbgower 
- I am not sure what the use case is for having a slider without a designated text input to show the value. The original intent of the slider component was to always have the input present. The slider component usage documentation still needs to be updated with our new content template, so that statement you mentioned will be revised.
- @carbon-design-system/developers Do you have any context as to why a `hideTextinput` prop was added to the component?

**Comment by abbeyhrt** [comment]
Looking at the history, `hideTextInput` has been there since the component's inception. My guess is it was added so that users could configure their own way of displaying the slider value outside of using that input, if they would like. While it is technically possible for a user to hide the input and not provide another way to see the value, I don't think that is how that prop `hideTextInput` is being used in the wild. We could definitely provide more clarification around that prop though! Something like "If true, you should/it is recommended to provide an alternative way to view the Slider's value"
